src/OAuthClient/BibClient.php

```diff
@@ -1,40 +1,77 @@
+use GuzzleHttp\Exception\ServerException;
 use NYPL\HoldRequestResultConsumer\Model\DataModel\Bib;
+use NYPL\HoldRequestResultConsumer\Model\Exception\NonRetryableException;
+use NYPL\HoldRequestResultConsumer\Model\Exception\RetryableException;
 use NYPL\Starter\APILogger;
 use NYPL\Starter\Config;
+use NYPL\Starter\Model\Response\ErrorResponse;
 
 class BibClient extends APIClient
 {
     /**
      * @param string $bibId
      * @param $nyplSource
-     * @return Bib | null
+     * @return null|Bib
+     * @throws NonRetryableException
+     * @throws RetryableException
      */
     public static function getBibByIdAndSource($bibId = '', $nyplSource)
     {
         $url = Config::get('API_BIB_URL') . '/'. $nyplSource . '/' . $bibId;
 
         APILogger::addDebug('Retrieving bib by Id and Source', (array) $url);
+        try {
+            $response = self::get($url);
 
-        $response = self::get($url);
+            $statusCode = $response->getStatusCode();
 
-        $response = json_decode((string) $response->getBody(), true);
+            $response = json_decode((string)$response->getBody(), true);
 
-        APILogger::addDebug(
-            'Retrieved bib by id and source',
-            $response['data']
-        );
+            APILogger::addDebug(
+                'Retrieved bib by id and source',
+                $response['data']
+            );
 
-        if ($response['statusCode'] !== 200) {
-            APILogger::addError(
-                'Failed',
-                array('Failed to retrieve bib ', $bibId, $response['type'], $response['message'])
+            // Check statusCode range
+            if ($statusCode === 200) {
+                return new Bib($response['data']);
+            } else {
+                APILogger::addError(
+                    'Failed',
+                    array('Failed to retrieve bib ', $bibId, $response['type'], $response['message'])
+                );
+                return null;
+            }
+        } catch (ServerException $exception) {
+            throw new RetryableException(
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                $exception->getResponse()->getStatusCode(),
+                null,
+                $exception->getResponse()->getStatusCode(),
+                new ErrorResponse(
+                    $exception->getResponse()->getStatusCode(),
+                    'internal-server-error',
+                    'Server Error from ' . __FUNCTION__ . $exception->getMessage()
+                )
+            );
+        } catch (ClientException $exception) {
+            throw new NonRetryableException(
+                'Client Error from '. __FUNCTION__ . $exception->getMessage(),
+                'Client Error from '. __FUNCTION__ . $exception->getMessage(),
+                $exception->getResponse()->getStatusCode(),
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.